configurations/dockerFileTemplate.go

```diff
@@ -8,57 +8,54 @@ import (
 )
 
 type dockerfileTemplate struct {
-	Data *dockerfileData
+	Data serverConfig
 }
 
 // Gives a new instance of dockerfile template.
-func newDockerfileTemplate(data *dockerfileData) *dockerfileTemplate {
+func newDockerfileTemplate(data serverConfig) *dockerfileTemplate {
 	return &dockerfileTemplate{Data: data}
 }
 
-// Unmarshal yaml file and gives an instance of dockerfileData.
-func newDockerFileDataFromYamlFile(filename string) (*dockerfileData, error) {
+// Unmarshal yaml file and gives an instance of serverConfig.
+func newDockerFileDataFromYamlFile(filename string) (serverConfig, error) {
 	node := yaml.Node{}
 
 	err := unmarshalYamlFile(filename, &node)
 	if err != nil {
 		return nil, fmt.Errorf("unmarshal: %v", err)
 	}
 
-	stages, err := getStagesDataFromNode(node.Content[0])
+	stageData, err := getStagesDataFromNode(node.Content[0])
 	if err != nil {
 		return nil, fmt.Errorf("can't extract Stages from node: %v", err)
 	}
-
-	return &dockerfileData{Stages: stages}, nil
+	return stageData, nil
 }
 
-// Gives all the stages and their corresponding data provided in yaml.
-func getStagesDataFromNode(node *yaml.Node) ([]stage, error) {
+// Returns the serverConfig instructions provided in yaml.
+func getStagesDataFromNode(node *yaml.Node) (serverConfig, error) {
 	var data dockerfileDataYaml
 
-	stagesInOrder, err := getStagesOrderFromYamlNode(node)
+	err := verifyStageYamlNode(node)
 	if err != nil {
 		return nil, fmt.Errorf("unmarshal: %v", err)
 	}
 	if err := node.Decode(&data); err != nil {
 		return nil, err
 	}
-	var stages []stage
-	for _, stageName := range stagesInOrder {
-		stages = append(stages, data.Stages[stageName])
-	}
-	return stages, nil
+	var stageData serverConfig
+
+	stageData = data.ServerConfig
```

Comment:
How is stageData the same as serverConfig?

Reply:
Changed